cvl/cvl_luascript.go

```diff
@@ -20,84 +20,40 @@
 package cvl
 
 import (
-	"github.com/Azure/sonic-mgmt-common/cvl/internal/util"
+	"errors"
+
 	"github.com/go-redis/redis/v7"
 )
 
-//Redis server side script
-func loadLuaScript(luaScripts map[string]*redis.Script) {
-
-	// Find entry which has given fieldName and value
-	luaScripts["find_key"] = redis.NewScript(`
-	local tableName=ARGV[1]
-	local sep=ARGV[2]
-	local keySetNames = {}
-	ARGV[3]:gsub("([^|]+)",function(c) table.insert(keySetNames, c) end)
-	local fieldName=ARGV[4]
-	local fieldValue=ARGV[5]
-
-	local entries = {}
-
-	-- Check if field value is part of key
-	if (#keySetNames == 1) then
-		-- field is only key
-		entries=redis.call('KEYS', tableName..sep..fieldValue)
-	elseif (keySetNames[#keySetNames] == fieldName) then
-		-- field is the last key
-		entries=redis.call('KEYS', tableName..sep.."*"..sep..fieldValue)
-	else
-		-- field is not the last key
-		entries=redis.call('KEYS', tableName.."*"..sep..fieldValue..sep.."*")
-	end
-
-	if (entries[1] ~= nil)
-	then
-		return entries[1]
-	else
-
-	-- Search through all keys for fieldName and fieldValue
-		local entries=redis.call('KEYS', tableName..sep.."*")
-
-		local idx = 1
-		while(entries[idx] ~= nil)
-		do
-			local val = redis.call("HGET", entries[idx], fieldName)
-            local valArrFld = redis.call("HGET", entries[idx], fieldName.."@")
-            if ((val == fieldValue) or (valArrFld == fieldValue))
-			then
-				-- Return the key
-				return entries[idx]
-			end
-
-			idx = idx + 1
-		end
-	end
+// RunLua is a temporary API to run a named lua script in ConfigDb. Script name
+// can be one of "count_entries" or "filter_entries".
+// TODO move the script definitions to DBAL and remove this API
+func RunLua(name string, args ...interface{}) (interface{}, error) {
+	script := luaScripts[name]
+	if script == nil {
```

Comment:
If a key is not found, the script variable is init to zero. 
You can check a key -
```
	script, ok := luaScripts[name]
	if !ok { . . . }
```